pkgs/stdenv/generic/setup.sh

```diff
@@ -341,22 +342,22 @@ declare -a allPlatOffsets=(-1 0 1)
 # implements.
 findInputs() {
     local -r pkg="$1"
-    local -ri hostOffset="$2"
```

Comment:
Does shellcheck say to not do this? It seems reasonable to set -i:

> The variable is to be treated as an integer; 
>  arithmetic evaluation is performed when the 
>  variable is assigned a value.

https://ss64.com/bash/declare.html

Reply:
This is the more controversial change. -i does not work at the moment with osh.
Apparently, -i is basically a no-op in bash. The recommended way to do arithmetic is to use `((...))` where the -i does not make a difference.
There was a longer discussion though between the creator of osh and zimbatm
oilshell/oil#864
(I tried to link it in the commit message)

the -i does not come from shellcheck and could be made to work with osh, it would be a matter of making a PR upstream. At the moment it does not, so I just removed it.
(perhaps the commit title could have been clearer)

Reply:
Here is more relevant things around this at http://mywiki.wooledge.org/BashGuide/Parameters
Specifically
```
However, in practice the use of declare -i is exceedingly rare. In large part, this is because it creates behavior that can be surprising to anyone trying to maintain the script, who misses the declare statement. Most experienced shell scripters prefer to use explicit arithmetic commands (with ((...)) or let) when they want to perform arithmetic. 
```

Reply:
Abandon all hopes of types, it's bash :D 

If I remember correctly, `-i` changes the parser on the fly. It also doesn't catch errors so it's not super useful. Eg: `declare -ri foo=bar && echo $foo` will output `0`